grommet/grommet
# [TS] Make ListProps and ListExtendedProps generic

#### What does this PR do?

This PR makes the `ListProps` and `ListExtendedProps` interfaces generic, adding a type parameter for the type of items in the `List`.

#### Where should the reviewer start?

There is only one file to review, so starting there is the best option. The `Form` component that implements a similar pattern, so there is some background context there if the reviewer is unfamiliar with generic types.

#### What testing has been done on this PR?

All unit tests were run using the pre-commit hook.

#### How should this be manually tested?

Providing a generic type to the `List` component should not cause any errors. Moreover, the user should be able to use the `List` component without specifying a generic type.

#### Any background context you want to provide?

All context is described in issue #5041

#### What are the relevant issues?

Resolves #5041

#### Do the grommet docs need to be updated?

No, default types have not been changed.

#### Should this PR be mentioned in the release notes?

Just under the general TypeScript umbrella is fine.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.
